lib/AcmeService.php

```diff
@@ -73,7 +73,7 @@ private function doRegister($email, $agreement = null) {
                 }
             }
 
-            yield new CoroutineResult(new Registration($payload->contact, $payload->agreement, $payload->authorizations, $payload->certificates));
+            yield new CoroutineResult(new Registration($response->getHeader("location"), $payload->contact, $payload->agreement, $payload->authorizations, $payload->certificates));
```

Comment:
`Message::getHeader` returns an `array` in Artax. Could be solved with: `list($location) = $response->getHeader("location");` and then pass `$location`. To have it really sane, we should probably check whether the header exists, otherwise we leak an internal Artax exception.

`$payload->agreement`, `$payload->authorizations` and `$payload->certificates` are all optional, so they need a check whether they exist, otherwise we end up with undefined property errors.